TravisCI is broken

Sometime after this build:
https://travis-ci.org/github/geopm/geopm/builds/739372769
this started happening:
https://travis-ci.org/github/geopm/geopm/builds/740258426

This has been happening with all TravisCI builds.  The log indicates a failure to build a dependency of one of our Python dependencies, ```kiwisolver```:
```
  100% |████████████████████████████████| 61kB 11.6MB/s 
Installing collected packages: pyparsing, pillow, kiwisolver, certifi, numpy, six, python-dateutil, cycler, matplotlib, setuptools, natsort, psutil, pytz, pandas, numexpr, mock, tables, pycparser, cffi, pyyaml, future, wrapt, typed-ast, lazy-object-proxy, astroid, toml, isort, mccabe, pylint
  Running setup.py install for kiwisolver ... error
    Complete output from command /opt/python/3.6.3/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qto3vi2x/kiwisolver/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-oa1606m6-record/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    running install
    running build
    running build_ext
    building 'kiwisolver' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/py
    gcc -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -g -fstack-protector --param=ssp-buffer-size=4 -Wformat -Werror=format-security -fPIC -I/tmp/pip-build-qto3vi2x/kiwisolver/.eggs/cppy-1.1.0-py3.6.egg/cppy/include -I. -I/opt/python/3.6.3/include/python3.6m -c py/kiwisolver.cpp -o build/temp.linux-x86_64-3.6/py/kiwisolver.o -std=c++11
    cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
    In file included from ./kiwi/constraint.h:11:0,
                     from ./kiwi/kiwi.h:9,
                     from py/kiwisolver.cpp:9:
    ./kiwi/expression.h:53:17: error: function ‘kiwi::Expression& kiwi::Expression::operator=(kiwi::Expression&&)’ defaulted on its first declaration with an exception-specification that differs from the implicit declaration ‘kiwi::Expression& kiwi::Expression::operator=(kiwi::Expression&&)’
         Expression& operator=(Expression&&) noexcept = default;
                     ^
    In file included from ./kiwi/constraint.h:13:0,
                     from ./kiwi/kiwi.h:9,
                     from py/kiwisolver.cpp:9:
    ./kiwi/strength.h:30:14: warning: ‘kiwi::strength::strong’ defined but not used [-Wunused-variable]
     const double strong = create( 1.0, 0.0, 0.0 );
                  ^
    ./kiwi/strength.h:32:14: warning: ‘kiwi::strength::medium’ defined but not used [-Wunused-variable]
     const double medium = create( 0.0, 1.0, 0.0 );
                  ^
    ./kiwi/strength.h:34:14: warning: ‘kiwi::strength::weak’ defined but not used [-Wunused-variable]
     const double weak = create( 0.0, 0.0, 1.0 );
                  ^
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/opt/python/3.6.3/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-qto3vi2x/kiwisolver/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-oa1606m6-record/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-build-qto3vi2x/kiwisolver/
You are using pip version 9.0.1, however version 20.2.4 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
The command "PYENV_VERSION=3.6 pip install --user -r ./scripts/requirements.txt" failed and exited with 1 during .
Your build has been stopped.
```

The error signature seems to be the same with all builds.  It's odd because the command that runs immediately previous to this one works fine:
```
$ PYENV_VERSION=2.7 pip install --user -r ./scripts/requirements.txt
```
But this fails:
```
$ PYENV_VERSION=3.6 pip install --user -r ./scripts/requirements.txt
```

So it's a combination of newer Python with GCC 4.8 that's causing the problem.  It also works fine with the clang build.

--- a/.travis.yml
+++ b/.travis.yml
@@ -36,7 +36,7 @@ compiler:
     - clang #todo version?
     - g++-4.8
 
-dist: trusty
+dist: bionic
 
 language: cpp
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -54,7 +54,9 @@ before_install:
     - pyenv global 3.6
 
 install:
+    - PYENV_VERSION=2.7 pip install --upgrade pip
     - PYENV_VERSION=2.7 pip install --user -r ./scripts/requirements.txt
+    - PYENV_VERSION=3.6 pip install --upgrade pip
     - PYENV_VERSION=3.6 pip install --user -r ./scripts/requirements.txt
     - ./configure --disable-fortran ${BUILD_TYPE} --enable-beta --prefix=$GEOPM_INSTALL_ROOT
 

--- a/.travis.yml
+++ b/.travis.yml
@@ -45,7 +45,6 @@ git:
     depth: false
 
 before_install:
-    - rvm install 2.0.0
     - git fetch origin --tags
     - git fetch origin $TRAVIS_BRANCH
     - ./autogen.sh

